plugins/test/unit/plugins/distributors/yum/test_publish.py

```diff
@@ -1379,45 +1387,65 @@ def setUp(self):
+        self.mocked_dt.today.return_value = datetime.datetime.fromtimestamp(1497605154)
+        self.mocked_mtime.side_effect = fake_mtime
 
-    def test_process_main(self):
+    def test_remove_old_repodata(self):
         """
-        Test that repoview tool was called with proper parameters.
+        Test that RemoveOldRepoData step is called with the correct filtered
+        repodata files to remove.
         """
-        publisher_conf = {}
+        def test_data_path():
+            return '%s/data' % os.path.dirname(inspect.getfile(test))
 
-        step = publish.RemoveOldRepodataStep('/foo')
-        step.config = publisher_conf
-        step.parent = mock.MagicMock()
-        step.parent.get_config.return_value.get.side_effect = publisher_conf.get
-
-        filter_old_repodata_mock = mock.Mock(side_effect=step.filter_old_repodata)
-        with mock.patch.multiple(
-            "pulp_rpm.plugins.distributors.yum.publish.RemoveOldRepodataStep",
-            remove_repodata_file=mock.DEFAULT,
-            filter_old_repodata=filter_old_repodata_mock
-        ) as mocked_step_dict:
-            step.process_main()
-            filter_old_repodata_mock.assert_called_with(
-                ['1abcde-primary.xml.gz', '0abcde-primary.xml.gz',
-                 '1abcde-primary.xml.gz', '0abcde-primary.xml.gz'])
-            mocked_step_dict["remove_repodata_file"].assert_called_with("1abcde-primary.xml.gz")
+        repo_root = os.path.join(test_data_path(), self._testMethodName)
+
+        self._init_publisher()
+        remove_step = publish.RemoveOldRepodataStep(repo_root)
+        remove_step.parent = self.publisher
+
+        metadata_file_path = os.path.join(repo_root, 'repodata', 'repomd.xml')
+
+        # file_paths under plugins/test/data/test_remove_old_repodata "to_keep"
+        metadata_file_locations = [
+            metadata_file_path,
+            os.path.join(repo_root, 'repodata', 'repomd-tmp.xml'),
+            os.path.join(repo_root, 'repodata', 'repomd.xml.asc'),
+            os.path.join(repo_root, 'repodata', 'repomd-tmp.xml.asc')
+        ]
+
+        self.publisher.repomd_file_context.metadata_file_path = metadata_file_path
+        self.publisher.repomd_file_context.metadata_file_locations = metadata_file_locations
+
+        with mock.patch(
+            'pulp_rpm.plugins.distributors.yum.publish.'
+            'RemoveOldRepodataStep.remove_repodata_file'
+        ) as mock_remove:
+            remove_step.process_main()
+            calls = mock_remove.mock_calls
+            # while repomd-tmp.xml and repomd-tmp.xml.asc have been patched with
+            # a modified timestamp, they are to be kept and should not be removed
+            expected_filenames = set([
+                'filelists-tmp.sqlite.bz2',
+                'primary-tmp.xml.gz',
+                '464c4a0ef57d030eff7d8700f8f26fd5c0b965449438cef3ea462adb21a055fa-tmp'])
+
+            self.assertEqual(len(calls), 3)
```

Comment:
this is what i get instead of 3
```
FAIL: Test that RemoveOldRepoData step is called with the correct filtered
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ipanova/pulp_development/pulp_rpm/plugins/test/unit/plugins/distributors/yum/test_publish.py", line 1437, in test_remove_old_repodata
    self.assertEqual(len(calls), 3)
AssertionError: 13 != 3

```